roles/common/tasks/populate-asm-disks.yml

```diff
@@ -1,62 +1,43 @@
-- name: Collect new disk facts
-  setup:
-    gather_subset:
-      - '!all'
-      - '!any'
-      - devices
+- name: Resolve blk_device paths to first partition paths if they exist
+  ansible.builtin.shell: >
+    set -o pipefail;
+    case "{{ item.blk_device }}" in
+      /dev/disk/by-id/*)
+        part_path="{{ item.blk_device }}"-part1;
+        ;;
+      /dev/mapper/*)
+        part_path="{{ item.blk_device }}"p1;
+        ;;
+      /dev/*)
```

Comment:
What's the best way to handle a path that _doesn't_ start with /dev?

Reply:
Could you please clarify why we need to handle this? Aren't block devices typically located in the /dev directory?

Reply:
From a user perspective, what would be the best way to handle a blk_device entry that doesn't start with dev?  Handle it like /dev/mapper and don't set a part_path?  Try and use the default readlink?  Or output a fatal error?

AFAICT, the implicit default case we have now is like /dev/mapper:  part_path is never set, so implicitly empty.